airflow/providers/snowflake/hooks/snowflake.py

```diff
@@ -135,7 +137,7 @@ def __init__(self, *args, **kwargs) -> None:
         self.schema = kwargs.pop("schema", None)
         self.authenticator = kwargs.pop("authenticator", None)
         self.session_parameters = kwargs.pop("session_parameters", None)
-        self.query_ids = []
+        self.query_ids: List[str] = []
```

Comment:
Is the `self.query_ids = []` needed in the `run()` method?

Reply:
I assume that in `run()` method this statement resets the list with ids.
So it keeps just last run ids. I think that's reasonable

Reply:
Agree. this makes the hook re-runnable/